mex-wholebodymodel/library/src/modelsetworldframe.cpp

```diff
@@ -32,98 +29,71 @@ using namespace mexWBIComponent;
+bool ModelSetWorldFrame::allocateReturnSpace(int a, mxArray *[])
 {
-    // Nothing to do really
-    return(true);
+  // Nothing to do really
+  return true;
 }
 
-bool ModelSetWorldFrame::compute(int nrhs, const mxArray* prhs[])
+bool ModelSetWorldFrame::compute(int nrhs, const mxArray *prhs[])
 {
-    if(mxGetM(prhs[1]) != 9 || mxGetN(prhs[1]) != 1 || mxGetM(prhs[2]) != 3 || mxGetN(prhs[2]) != 1 || mxGetM(prhs[3]) != 3 || mxGetN(prhs[3]) != 1)
-    {
-        mexErrMsgIdAndTxt( "MATLAB:mexatexit:invalidNumInputs","Malformed state dimensions/components");
-    }
-
-    double *R_temp,*p_temp, *g_temp;
-    R_temp = (double *)mxGetPr(prhs[1]);
-    p_temp = (double *)mxGetPr(prhs[2]);
-    g_temp = (double *)mxGetPr(prhs[3]);
-
-    double tempR[9],tempP[3];
-
-    for(int i = 0;i<3;i++)
-    {
-        tempP[i] = p_temp[i];
-    }
-    reorderMatrixElements(R_temp, tempR);
-
-    wbi::Rotation tempRot(tempR);
-    wbi::Frame tempFrame(tempRot, tempP);
-    modelState->setRootWorldRotoTranslation(tempFrame);
-    modelState->setGravity(g_temp);
-    return true;
+  if( mxGetM(prhs[1]) != 9 || mxGetN(prhs[1]) != 1 || mxGetM(prhs[2]) != 3 ||
+      mxGetN(prhs[2]) != 1 || mxGetM(prhs[3]) != 3 || mxGetN(prhs[3]) != 1 )
+  {
+    mexErrMsgIdAndTxt("MATLAB:mexatexit:invalidNumInputs", "Malformed state dimensions/components");
+  }
+  double *R_temp, *p_temp, *g_temp;
+  R_temp = mxGetPr(prhs[1]);
+  p_temp = mxGetPr(prhs[2]);
+  g_temp = mxGetPr(prhs[3]);
+
+  double tempR[9];
+  reorderMatrixInRowMajor(R_temp, tempR);
+  wbi::Rotation tempRot(tempR);
+  wbi::Frame tempFrame(tempRot, p_temp);
+
+  modelState->setRootWorldRotoTranslation(tempFrame);
+  modelState->setGravity(g_temp);
+
+  return true;
 }
 
 bool ModelSetWorldFrame::computeFast(int nrhs, const mxArray* prhs[])
 {
-    if(mxGetM(prhs[1]) != 9 || mxGetN(prhs[1]) != 1 || mxGetM(prhs[2]) != 3 || mxGetN(prhs[2]) != 1)
-    {
-        mexErrMsgIdAndTxt( "MATLAB:mexatexit:invalidNumInputs","Malformed state dimensions/components");
-    }
-
-    double *R_temp,*p_temp;
-    R_temp = (double *)mxGetPr(prhs[1]);
-    p_temp = (double *)mxGetPr(prhs[2]);
-
-
-    double tempR[9],tempP[3];
-    for(int i = 0;i<9;i++)
-    {
-        tempR[i] = R_temp[i];
-        if(i<3)
-        {
-            tempP[i] = p_temp[i];
-        }
-    }
-    wbi::Rotation tempRot(tempR);
-    wbi::Frame tempFrame(tempRot, tempP);
-    //H_baseLink_wrWorld = temp;
-    //H_baseLink_wrWorld = tempFrame;
-    modelState->setRootWorldRotoTranslation(tempFrame);
-    //    mexPrintf("Roto translation of world from base frame applied \n");
-
-    return true;
-}
-
+  if( mxGetM(prhs[1]) != 9 || mxGetN(prhs[1]) != 1 || mxGetM(prhs[2]) != 3 || mxGetN(prhs[2]) != 1 )
+    mexErrMsgIdAndTxt("MATLAB:mexatexit:invalidNumInputs", "Malformed state dimensions/components");
 
+  double *R_temp, *p_temp;
+  R_temp = mxGetPr(prhs[1]);
+  p_temp = mxGetPr(prhs[2]);
 
+  wbi::Rotation tempRot(R_temp);
```

Comment:
I know that Matlab and also Fotran are using column-major ordering. Here in this part of the code I did not changed much. Only some optimizations and the replacement of the double for-loop with the self-made transposition-function. But I guess that in the old code of the compute-fast function there is also missing the transposition-function. So in this case we have found an old hidden bug. I will check it and repair it.